### PR TITLE
Do not rely on boot module loader for finding module dependencies

### LINKED ISSUE #28448: Avoid deprecated `jboss-modules` method usage
### Description

The `Module.getBootModuleLoader()` API is slated for deprecation and eventual removal. Its usage is not strictly necessary; it can be replaced using `Module.getCallerModuleLoader()` which has several advantages (particularly around embedding WildFly or other scenarios where class loaders my be recreated).

The existing tests should cover this change.

### PR DESCRIPTION
**Pull request by dmlloyd**
The `Module.getBootModuleLoader()` API may be deprecated at some point soon, so use a safer alternative.

The existing tests should cover this change.

Closes #28448.

### CODE REVIEW

**Review by jamezp — APPROVED** [review]

**Review by pskopek — APPROVED** [review]
Looks good to me.

**Review by pskopek — APPROVED** [review]

**Comment by dmlloyd** [comment]
Still waiting for maintainer feedback. If nobody has an objection, could we move it forward?

**Comment by ahus1** [comment]
@dmlloyd - I've now started the builds, let's see if they are green. 

To get it merged, could you please create an issue and update the description to references it with `Closes #xxxx` as described in the CONTRIBUTING.md? 

With the approvals already in, it should then be ready to be merged.

**Comment by dmlloyd** [comment]
Done, thanks!

**Review by ahus1 — APPROVED** [review]
Approving based on @pskopek's review. Thank you for this PR!

**Pull request merged**
